### DISCUSSION TITLE
@this.call not working as expected

### DISCUSSION THREAD

**Discussion by hyphen81** [livewire/livewire#8848] [1. Help]
I have some code that needs to live in javascript and communicate back to the server. I'm using `@this.call` to handle that. Side note/question, I noticed that's not in the v3 documentation, so is there a different way I should be doing it now?

I'm essentially calling the server method to handle the validation, and based on the results of the validation, I want to do different things. The validation method is correctly kicking back the ValidationException, and I can see the failed validation fields in the UI, but `.then()` is still called and runs, presumably because the response is a 200 response.

Here's the simplified code:

```
@this.call('doStuff', params).then(() => {
//do stuff only if successful
}).catch((error) => {
//handle the error (validation exception)
});
```

I've tried this too and `response` is null.
```
@this.call('doStuff', params).then((response) => {
    console.log(response)
}).catch((error) => {

});
```

**Comment by EdgarsJoja** [comment]
I think doing this using `$wire` object is how it is meant to be done in Livewire.
See: https://livewire.laravel.com/docs/actions#calling-actions-from-alpine

However, I'm not sure if this would eliminate the root issue here (promise not being rejected).
